c/src/platform/platform.h

```diff
@@ -65,7 +65,7 @@ int pn_i_error_from_errno(pn_error_t *error, const char *msg);
  */
 int64_t pn_i_atoll(const char* num);
 
-int pni_snprintf(char *buf, size_t count, const char *fmt, ...);
+PN_EXTERN int pni_snprintf(char *buf, size_t count, const char *fmt, ...);
```

Comment:
Can't do this pni_ prefix means internal only not to be exported.
If you need to use this in a different library you need to copy the code.

Reply:
I saw two `pni_` symbols in qpid-proton-cored.lib already, so I did it this way. Ok, I'll link it directly.

    dumpbin /EXPORTS qpid-proton\out\build\x64-Debug\c\qpid-proton-cored.lib

    [...]

                  pn_void_refcount
                  pn_work_head
                  pn_work_next
                  pni_message_get_extra
                  pni_message_with_extra
                  pnx_sasl_clear_password
                  pnx_sasl_error
                  pnx_sasl_fail_authentication
                  pnx_sasl_get_allow_insecure_mechs
                  pnx_sasl_get_auth_required
                  pnx_sasl_get_context
                  pnx_sasl_get_external_ssf
    [...]

Reply:
Yes, those symbols are the subject of an issue (PROTON-1637) they should either be public API with a well defined meaning or not exported.
pni == ProtoN Internal.

Reply:
I've made the change. Is this what you've meant? Just realized I'd have to do the same thing to `qpid-proton-proactor-static` if I wanted to go this way.